### PR TITLE
C++ algorithms

### PR DESCRIPTION
**Pull request by vatikagupta**

### CODE REVIEW

File: C++ Algorithms/lcsa.cpp [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	}
+	else if(*((p+i*n) + j) == '-'){
+		printlcs(n, p, A, i, j-1);
+	}
+}
+
+void lcs(string A, string B, int m, int n){
+	//Function to calculate longest common subsequence
+	int l[m+1][n+1], i, j;
+	char p[m+1][n+1];
```
Comment (khanaaziz1) [inline C++ Algorithms/lcsa.cpp:26]: recheck these two lines. final code should be working fine
tip: m and n should be constant or you can use dynamic array

**Review by khanaaziz1 — CHANGES_REQUESTED** [review]
following changes required

File: C++ Algorithms/lcsa.cpp (hunk 1 above)
Comment (khanaaziz1) [reply C++ Algorithms/lcsa.cpp:26]: you can use this example:

        int arr[10][10];
	int x = 10;
	int** l = new int *[x];
	l[m + 1][n + 1];
	int i, j;
	char** p = new char *[x];
	p[m + 1][n + 1];

Comment (vatikagupta) [reply C++ Algorithms/lcsa.cpp:26]: The code is working fine as variables m and n get their values via function arguments. So they are in a sense constants during array declaration.
Output

**Pull request merged**